NoteBerkayMvvm/Scenes/Notes/AddNote/AddNoteViewController.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  AddNoteViewController.swift
+//  NoteBerkayMvvm
+//
+//  Created by Mustafa Berkay Kaya on 24.02.2022.
+//
+
+import UIKit
+import MobilliumBuilders
+import TinyConstraints
+
+final class AddNoteViewController: BaseViewController<AddNoteViewModel> {
+    
+    private let mainStackView = UIStackViewBuilder()
+        .spacing(19)
+        .axis(.vertical)
+        .distribution(.fill)
+        .build()
+    private let titleTextField = AuthTextField()
+    private let descriptionTextView = UITextViewBuilder().build()
+    private let saveNoteButton = CustomButton()
+    var noteID: Int = 0
+    var type: DetailVCShowType = .add
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+      
+        titleTextField.text = viewModel.titleText
+        descriptionTextView.text = viewModel.descriptionText
+        noteID = viewModel.noteId
+        type = viewModel.type
+        
+        configureContents()
+        addSubViews()
+    }
+    
+}
+// MARK: - UILayout
+extension AddNoteViewController {
+    private func addSubViews() {
+        descriptionTextView.delegate = self
+        switch type {
+        case .showNote:
+            addMainStackView()
+            addTitleNote()
+            addDescriptionTextView()
```

Comment:
bunlar tüm typelar için ortak dışarı alınabilir ve case içine sadece farklılıklar konabilir.